common/http/src/batch.rs

```diff
@@ -0,0 +1,630 @@
+        ];
+
+        let stream0 = stream::iter(input0.iter());
+
+        let input1 = vec![OffsetLine::new(
+            Line::builder().line("4".to_string()).build().unwrap(),
+        )];
+        let stream1 = stream::iter(input1.iter())
+            .then(move |n| Delay::new(Duration::from_millis(300)).map(move |_| n));
+
+        let input2 = vec![
+            OffsetLine::new(Line::builder().line("5".to_string()).build().unwrap()),
+            OffsetLine::new(Line::builder().line("6".to_string()).build().unwrap()),
+            OffsetLine::new(Line::builder().line("7".to_string()).build().unwrap()),
+            OffsetLine::new(Line::builder().line("8".to_string()).build().unwrap()),
+            OffsetLine::new(Line::builder().line("9".to_string()).build().unwrap()),
+        ];
+        let stream2 = stream::iter(input2.iter());
+
+        let stream = stream0.chain(stream1).chain(stream2);
+
+        let batch_stream = stream.timed_request_batches(5000, Duration::from_millis(100));
+
+        let now = Instant::now();
+        let min_times = [Duration::from_millis(80), Duration::from_millis(150)];
+        let max_times = [Duration::from_millis(350), Duration::from_millis(500)];
+
+        let mut i = 0;
+
+        let results = batch_stream
+            .map(move |s| {
+                let now2 = Instant::now();
+                assert!((now2 - now) < max_times[i]);
+                assert!((now2 - now) > min_times[i]);
+                i += 1;
+                s
+            })
+            .collect::<Vec<_>>()
+            .await;
+
+        let mut buf = String::new();
+        results[0]
+            .as_ref()
+            .unwrap()
+            .0
+            .reader()
+            .read_to_string(&mut buf)
+            .unwrap();
+        let mut body: HashMap<String, Vec<Line>> = serde_json::from_str(&buf).unwrap();
+        let lines0: Vec<OffsetLine> = body
+            .remove("lines")
+            .unwrap_or_default()
+            .into_iter()
+            .map(OffsetLine::new)
+            .collect();
+        assert_eq!(
+            input0.len(),
+            lines0.len(),
+            "input {:#?}\nresult {:#?}",
+            input0,
+            lines0
+        );
+        assert_eq!(input0, lines0);
+
+        let mut expected = input1.clone();
+        expected.extend_from_slice(&input2[..]);
+
+        buf.clear();
+        results[1]
+            .as_ref()
+            .unwrap()
+            .0
+            .reader()
+            .read_to_string(&mut buf)
+            .unwrap();
+        let mut body: HashMap<String, Vec<Line>> = serde_json::from_str(&buf).unwrap();
+        let lines1: Vec<OffsetLine> = body
+            .remove("lines")
+            .unwrap_or_default()
+            .into_iter()
+            .map(OffsetLine::new)
+            .collect();
+
+        assert_eq!(expected, lines1);
+    }
+
+    proptest! {
+        #[test]
+        fn roundtrip(
+            inp in (0..512usize)
```

Comment:
This property test takes around 3 mins on my local machine, can we use a smaller set?

Reply:
Yeh, I've been running it in release mode, 3 mins is excessive. I'll change the number of iterations